src/io_scheduler.cpp

```diff
@@ -323,8 +296,8 @@ auto io_scheduler::process_scheduled_execute_inline() -> void
         tasks.swap(m_scheduled_tasks);
 
         // Clear the schedule eventfd if this is a scheduled task.
-        eventfd_t value{0};
-        eventfd_read(m_schedule_fd, &value);
+        int control = 0;
+        ::read(m_schedule_fd[1], reinterpret_cast<void*>(&control), sizeof(control));
```

Comment:
codacy is complaining the return value of `::read` isn't used, but I never checked `eventfd_read` before either, I think this is reasonably safe since we guarantee we've written to it via the `m_schedule_fd_triggered` atomic.

Reply:
I'll resolve this so you can merge it, but just wanted to make a note of it.